src/backends/c.rs

```diff
@@ -108,6 +108,8 @@ impl CodeGen for CGenContext {
 }
 
 impl CGenContext {
+    pub const C_OUTPUT_FILENAME: &'static str = "out.c";
```

Comment:
Previous behavior, can change this if you want it and the IR file to be dynamic

Reply:
Looks good to me! I don't think there is a huge need to make the intermediate files have customizable names for now, since debugging is their main use case at this point.